enhancements/microshift/optionally-deploy-csi-plugin.md

```diff
@@ -0,0 +1,256 @@
+space on the MicroShift host.
+
+## Motivation
+
+Not all users may want dynamic persistent storage or volume snapshotting for their MicroShift deployments and should be
+enabled to choose whether to deploy the driver and / or snapshotter. This will afford resource-conscious users the
+opportunity to better tune their MicroShift deployment to their hardware requirements.
+
+### User Stories
+
+- A user deploys MicroShift with the LVMS driver and CSI snapshot controller.
+- A user deploys MicroShift with the LVMS driver only.
+- A user deploys MicroShift with neither the LVMS driver nor the CSI snapshot controller.
+
+### Goals
+
+- Provide the LVMS CSI driver and LVMS CSI snapshot controller as optional components of MicroShift
+- Reuse the optional-installation pattern implemented by MicroShift for the [Multus CNI
+  Plugin](./multus-cni-for-microshift.md).
+- Enable installing the LVMS CSI driver and LVMS CSI snapshot controller to a cluster that was previously deployed
+  without the components.
+
+### Non-Goals
+
+- Generalizing this design to install any other CSI driver.
+- Uninstalling the CSI driver or snapshotter rpms will not uninstall the cluster level components. This is a data safety
+  concern that should be handled by the user.
+- Support installing CSI snapshotter, but not the CSI driver. The snapshotter depends on the CSI driver; it cannot
+  run as a standalone component.
+
+## Proposal
+
+Extract away deployment of the LVMS CSI driver and LVMS CSI snapshot controller from MicroShift logic and provide the
+components as separate, optional rpms. The new rpms will be named `microshift-lvms` and `microshift-lvms-snapshotting`.
+`microshift-lvms-snapshotting` will have a dependency on `microshift-lvms` because it cannot be run effectively as a
+standalone component. Users should be able to install these rpms simultaneously with core MicroShift rpms. It should
+also be possible to install LVMS and CSI components to a cluster that was initially deployed without them. This should
+not be a reversible process: uninstalling the rpms should not delete CSI components or cluster APIs from the cluster.
+Doing so endangers users' data and could lead to orphaning of LVM volumes. Instead, uninstalling the rpms will only
+ensure that LVMS and Snapshotting are not deployed on MicroShift startup/restart. It will be the user's responsibility
+to ensure there are not LVMS PVs in the cluster before deleting the LVMS and Snapshotting API instances.
+
+### Workflow Description
+
+**_Prerequisites_**
+
+- Host has LVM installed
+
+**_Installation with CSI Driver and Snapshotting_**
+
+1. User determines there is a requirement for persistent storage and volume snapshotting.
+2. User specifies an ostree blueprint which includes the following sections:
+    1. Packages: microshift, microshift-greenboot, microshift-networking, microshift-selinux, microshift-lvms, 
+       microshift-lvms-snapshotting
+    2. File (Optional): lvmd.yaml
```

Comment:
Are we planning to install a default lvmd.yaml file? If so, specifying it in the blueprint would override the default. 
Let's clarify this please